pkg/k8sutil/k8sutil.go

```diff
@@ -254,12 +254,16 @@ func mergeMaps(new map[string]string, old map[string]string) map[string]string {
 }
 
 func mergeKubectlAnnotations(new *metav1.ObjectMeta, old metav1.ObjectMeta) {
-	new.SetAnnotations(mergeMapsByPrefix(new.Annotations, old.Annotations, "kubectl.kubernetes.io/"))
+	new.SetAnnotations(mergeMapsByPrefix(old.Annotations, new.Annotations, "kubectl.kubernetes.io/"))
```

Comment:
The `old` and `new` parameters are mixed up here compared to the function declaration and it's a bit confusing.

Reply:
Yes, that could be confusing. I swapped them here instead: https://github.com/prometheus-operator/prometheus-operator/pull/4422/files#diff-0128909c1627157f3265add5bfeacace324a45ac2aa7c81bf9e91e887edf67faR171.

We could also rename `old` and `new` into `from` and `to` to make it more clear what is being merged into what.